frontend/alert/src/components/AutoComplete.js

```diff
@@ -0,0 +1,237 @@
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faHistory } from "@fortawesome/free-solid-svg-icons";
+import { useOnClickOutside } from "./shared/useOnClickOutside";
+import { Input } from "./Input";
+
+/* A function that takes in a search term and returns a promise with both the search term and
+the search results.
+Including the search term makes it possible to determine if the search result is stale.
+ */
+const suggestionsFor = search => fetch(`/api/alert/courses?search=${search}`)
+    .then(res => res.json()
+        .then(searchResult => ({
+            searchResult,
+            searchTerm: search,
+        })));
+
+/* Debounce the search promise so that it doesn't make requests to the backend more frequently
+than the given interval
+*/
+const SUGGESTION_INTERVAL = 250;
+const suggestionsDebounced = AwesomeDebouncePromise(
+    suggestionsFor,
+    SUGGESTION_INTERVAL,
+);
+
+const AUTOCOMPLETE_BORDER_WIDTH = 1;
+
+const SuggestionsContainer = styled.div`
+    position: absolute;
+    left: 0;
+    top: 100%;
+    width: ${({ below }) => below && (below.getBoundingClientRect().width - AUTOCOMPLETE_BORDER_WIDTH * 2)}px;
+    visibility: ${({ hidden }) => (hidden ? "hidden" : "visible")};
+    z-index:5000;
+    text-align: left;
+`;
+
+const SuggestionsBox = styled.div`
+    padding-bottom: 0.5rem;
+    border-color: #d6d6d6;
+    border-width: ${AUTOCOMPLETE_BORDER_WIDTH}px;
+    margin-top: 0.7rem;
+    margin-right: 0;
+    width: 100%;
+    border-style: solid;
+    background-color: white;
+    padding-top: 0;
+    max-height: 20rem;
+    overflow-y: scroll;
+    &::-webkit-scrollbar {
+        display: none;
+    }
+`;
+
+const SuggestionBox = styled.div`
+    border-bottom-style: solid;
+    border-width: 1px;
+    border-color: #d6d6d6;
+    padding-left: 0.5rem;
+    padding-bottom: 1rem;
+    display: flex;
+    justify-content: stretch;
+    flex-direction: row;
+    cursor: pointer;
+    transition: 180ms ease background;
+    &:hover {
+        background-color: rgb(220, 220, 220);
+    }
+`;
+
+const SuggestionTitle = styled.div`
+   color: #282828;
+   font-size: 1rem;
+   font-family: Inter Medium;
+   padding-top: 0.5rem;
+`;
+
+const SuggestionSubtitle = styled.div`
+   color: #282828;
+   font-size: 0.9rem;
+   padding-top: 0.4rem;
+   font-family: Inter Regular;
+`;
+
+const IconContainer = styled.div`
+    display: flex;
+    flex-direction: column;
+    justify-content: center;
+    align-items: center;
+    flex-grow: 1;
+    flex-basis: 20%;
+`;
+
+const SuggestionLeftCol = styled.div`
+   max-width: 80%;
+   flex-basis: 80%;
+   flex-grow: 1;
+`;
+
+const AutoCompleteInput = styled(Input)`
+    position: absolute;
+    background: transparent;
+    z-index: 1;
+`;
+
+const AutoCompleteInputBackground = styled(AutoCompleteInput)`
+    background: white;
+    z-index: 0;
+    color: grey;
+`;
+
+const AutoCompleteContainer = styled.div`
```

Comment:
This is more of a TODO for after an MVP but we should look to be generalizing divs like this into a `Div` component where we can set different things by props.